core/field.js

```diff
@@ -604,3 +604,14 @@ Blockly.Field.prototype.setTooltip = function(_newTip) {
 Blockly.Field.prototype.getAbsoluteXY_ = function() {
   return goog.style.getPageOffset(this.borderRect_);
 };
+
+/**
+ * Whether this field references any Blockly variables.  If true it may need to
+ * be handled differently during serialization and deserialization.  Subclasses
+ * may override this.
+ * @return {boolean} True if this field has any variable references.
+ * @package
+ */
+Blockly.Field.prototype.referencesVariables = function() {
```

Comment:
This name bothers me, as I have a hard time parsing it in context as a boolean.  The best alternative I can think of is `.isVariablerReference()`.  (Obviously my Java background showing through.) Feel free if to ignore if you disagree.